src/libraries/System.Runtime/tests/System/Reflection/InvokeWithRefLikeArgs.cs

```diff
@@ -0,0 +1,160 @@
+
+using Moq;
+using Xunit;
+using Xunit.Sdk;
+
+namespace System.Reflection.Tests
+{
+    public class InvokeWithRefLikeArgs
+    {
+        [Fact]
+        public static void MethodReturnsRefToRefStruct_ThrowsNSE()
+        {
+            MethodInfo mi = GetMethod(nameof(TestClass.ReturnsRefToRefStruct));
+            Assert.Throws<NotSupportedException>(() => mi.Invoke(null, null));
+        }
+
+        [Fact]
+        public static void MethodTakesRefStructAsArg_DoesNotCopyValueBack()
+        {
+            MethodInfo mi = GetMethod(nameof(TestClass.TakesRefStructAsArg));
+
+            object[] args = new object[] { null };
+            mi.Invoke(null, args);
+
+            Assert.Null(args[0]); // no value should have been copied back
+        }
+
+        [Fact]
+        public static void MethodTakesRefStructAsArgWithDefaultValue_DoesNotCopyValueBack()
+        {
+            MethodInfo mi = GetMethod(nameof(TestClass.TakesRefStructAsArgWithDefaultValue));
+
+            object[] args = new object[] { Type.Missing };
+            mi.Invoke(null, args);
+
+            Assert.Null(args[0]); // no value should have been copied back
+        }
+
+        [Fact]
+        public static void MethodTakesRefToRefStructAsArg_ThrowsNSE()
+        {
+            // Use a Binder to trick the reflection stack into treating the returned null
+            // as meaning "use the default value of the ref struct".
+
+            Mock<Binder> mockBinder = new Mock<Binder>(MockBehavior.Strict);
+            Type myRefStructType = typeof(MyRefStruct);
+            mockBinder.Setup(o => o.ChangeType("hello", myRefStructType.MakeByRefType(), null)).Returns((object)null);
+
+            MethodInfo mi = GetMethod(nameof(TestClass.TakesRefToRefStructAsArg));
+            Assert.Throws<NotSupportedException>(() => mi.Invoke(null, BindingFlags.InvokeMethod, mockBinder.Object, new object[] { "hello" }, null));
+        }
+
+        [Fact]
+        public static void MethodTakesOutToRefStructAsArg_ThrowsNSE()
+        {
+            MethodInfo mi = GetMethod(nameof(TestClass.TakesOutToRefStructAsArg));
+            Assert.Throws<NotSupportedException>(() => mi.Invoke(null, new object[] { null }));
+        }
+
+        [Fact]
+        public static void PropertyTypedAsRefToRefStruct_AsMethodInfo_ThrowsNSE()
+        {
+            MethodInfo mi = GetMethod("get_" + nameof(TestClass.PropertyTypedAsRefToRefStruct));
+            Assert.Throws<NotSupportedException>(() => mi.Invoke(null, null));
+        }
+
+        [Fact]
+        public static void PropertyTypedAsRefToRefStruct_AsPropInfo_ThrowsNSE()
+        {
+            PropertyInfo pi = typeof(TestClass).GetProperty(nameof(TestClass.PropertyTypedAsRefToRefStruct));
+            Assert.NotNull(pi);
+            Assert.Throws<NotSupportedException>(() => pi.GetValue(null));
+        }
+
+        [Fact]
+        public static void PropertyIndexerWithRefStructArg_DoesNotCopyValueBack()
```

Comment:
This worked previously, correct?

Reply:
Yeah. It's adding missing test coverage.